tests/spread/deploy-basic/task.yaml

```diff
@@ -0,0 +1,62 @@
+summary: Run basic deployment tests
+kill-timeout: 90m
+systems:
+  - ubuntu-24.04
+
+prepare: |
+  . "${CRAFT_TEST_PATH}/common.sh"
+  setup_test_environment
+
+execute: |
+  pushd "$PROJECT_PATH"
+
+  # Build the charm if it doesn't already exist
+  if [[ ! -f "$PWD/zinc-k8s_amd64.charm" ]]; then
+    charmcraft pack --verbose
+  else
+    chown $(id -u):$(id -g) $PWD/zinc-k8s_amd64.charm
+  fi
```

Comment:
you can move this to prepare, as it is not really what you are testing, not going to get into a philosophical debate though :-D
This can go into the `spread.yaml` _top-level_ or _suite-level_ `prepare` to avoid repetition

Reply:
Oh nice, good idea!